src/ui/buttons/props.js

```diff
@@ -696,6 +719,66 @@ export function normalizeButtonStyle(
   };
 }
 
+export function normalizeButtonMessage(
+  props: ?ButtonPropsInputs,
+  message: ButtonMessageInputs
+): ButtonMessage {
+  if (!message) {
+    throw new Error(`Expected props.message to be set`);
+  }
+
+  const { amount, offer, color, position, align } = message;
+
+  if (typeof amount !== "undefined") {
+    if (typeof amount !== "number") {
+      throw new TypeError(
+        `Expected message.amount to be a number, got: ${amount}`
+      );
+    }
+    if (amount < 0) {
+      throw new TypeError(
+        `Expected message.amount to be a positive number, got: ${amount}`
+      );
```

Comment:
This should likely be an `Error` instead of `TypeError` since we're validating the value.